aws_lambda_powertools/utilities/data_classes/common.py

```diff
@@ -48,7 +48,9 @@ def get_query_string_value(self, name: str, default_value: Optional[str] = None)
         params = self.query_string_parameters
         return default_value if params is None else params.get(name, default_value)
 
-    def get_header_value(self, name: str, default_value: Optional[str] = None) -> Optional[str]:
+    def get_header_value(
```

Comment:
Shouldn't this be available only for where there are `self.headers` instead of being a common method  
e.g. API GW?

Reply:
> Shouldn't this be available only for where there are `self.headers` instead of being a common method
> e.g. API GW?

`BaseProxyEvent` is the base class for `APIGatewayProxyEvent`, `APIGatewayProxyEventV2`, and `ALBEvent` which all support `self.headers`. See test events `apiGatewayProxyEvent.json`, `apiGatewayProxyV2Event.json` and `albEvent.json`.

Reply:
@heitorlessa ☝️  - does this make sense?

Reply:
It does now, thanks for clarifying! Could you then add that in the docstring so one doesn't inadvertently use it for AppSync, for example?